ethereum/gateway-eth-ts/src/GatewayTs.ts

```diff
@@ -45,19 +47,41 @@ export class GatewayTs extends GatewayTsBase {
     return ethTransaction(contract, 'removeNetworkAuthority', args, options);
   }
 
-  async issue(owner: string, tokenId: number | BigNumber = null, expiration: number | BigNumber = 0, bitmask: BigNumber = BigNumber.from('0'), constrains: BigNumber = BigNumber.from('0'), gatewayTokenAddress?: string, options?: TxOptions):Promise<string> {
-    const gatewayToken = this.getGatewayTokenContract(gatewayTokenAddress);
+  async issue(
+    owner: string, 
+    tokenId: number | BigNumber = null, 
+    expiration: number | BigNumber = 0, 
+    bitmask: BigNumber = BigNumber.from('0'), 
+    constrains: BigNumber = BigNumber.from('0'), 
+    gatewayTokenAddress?: string, 
+    options?: TxOptions): Promise<SendableTransaction> {
+    
+    const gatewayTokenContract = this.getGatewayTokenContract(gatewayTokenAddress);
 
     if (tokenId === null) {
-      tokenId = await this.generateTokenId(owner, constrains, gatewayToken);
+      tokenId = await this.generateTokenId(owner, constrains, gatewayTokenContract);
     }
 
     if (expiration > 0) {
       expiration = getExpirationTime(expiration);
     }
 
-    const args: any[] = [owner, tokenId, expiration, bitmask];
-    return ethTransaction(gatewayToken.contract, 'mint', args, options);
+    const txRequest = await populateTx(gatewayTokenContract.contract, 'mint', [owner, tokenId, expiration, bitmask], options);
+
+    if (options?.forwardTransaction) {
+      const { request, signature } = await signMetaTxRequest(this.wallet, this.forwarder.contract, {
+        from: this.wallet.address,
+        to: gatewayTokenContract.contract.address,
+        data: txRequest.data
+      });
+
+      const metaTx = await populateTx(this.forwarder.contract, 'execute', [request, signature], options);
+      const sendableTransaction = new SendableTransaction(gatewayTokenContract.contract, metaTx, options);
```

Comment:
line duplication here between 78,79 and 83, 84. You can just create the tx/metatx and  then pass it into the final line which should just be return new SendableTransaction...:

So something like

```
const txRequest = await populateTx(...)
const txToSend = wrapTxIfForwarded(txRequest, options, ...)
return new SendableTransaction(..., txToSend,...)
```

Note also, this pattern applies to _all transactions_, not just issue, so the wrapTxIfForwarded (or whatever you choose to name it) should not be specific to issue. The logic for signing and calling the forwarder should not be in issue().